src/routing_membrane.rs

```diff
@@ -524,10 +527,33 @@ impl<F> RoutingMembrane<F> where F: Interface {
 
         // Handle FindGroupResponse
         match  message.message_type {
-            MessageType::FindGroupResponse(ref vec_of_public_ids) =>
+            MessageType::FindGroupResponse(ref vec_of_public_ids) => {
+                let mut quorum = types::QUORUM_SIZE;
+                if self.routing_table.size() < types::QUORUM_SIZE {
+                    quorum = self.routing_table.size();
+                }
+
+                let source = match message.source.actual_source() {
+                    Address::Node(source) => source,
+                    _ => return Err(RoutingError::Response(ResponseError::InvalidRequest)),
+                };
+
+                let request = SentinelFindGroupResponse::new(message.source.non_relayed_source(), message.message_id);
+                let resolved = match self.find_group_response_sentinel.add_identities(
+                        request.clone(),
+                        source.clone(),
+                        message_wrap.encoded_body().clone(),
+                        message_wrap.signature().clone(),
+                        PublicIdGroupClaim { identities: vec_of_public_ids.clone() },
+                        quorum) {
+                    Some((request, vec_of_public_ids)) => (request, vec_of_public_ids),
+                    None => return Ok(())
+                };
+
                 ignore(self.handle_find_group_response(
-                            vec_of_public_ids.clone(),
-                            address_in_close_group_range.clone())),
+                            resolved.1.clone(),
+                            address_in_close_group_range.clone()));
+            },
```

Comment:
Yes, sorry, missed the fact we have the function already. Then, what I was trying to ask is whether you could move the added lines to the `handle_find_group_response` function's body. It would be consistent with the other `handle_` functions and we would have less code inside this (already quite big) `message_received` function.